### PR TITLE
PD-1828: Form footer style updates

### PR DESCRIPTION
**Pull request by spark3-mdb**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [PD-1828](https://jira.mongodb.org/browse/PD-1828)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Style change (purely visual updates; if this is the case, attach a screenshot below!)

image
image
image
image
image

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/form-footer/src/FormFooter.tsx [hunk 1]
```diff
@@ -119,60 +114,41 @@ export default function FormFooter({
...
-        <span className={flexSpan}>
-          {backButtonText && (
+        )}
+        <div className={flexEndContent}>
+          {errorMessage && (
+            <Banner className={bannerStyle} variant="danger">
+              {errorMessage}
+            </Banner>
+          )}
+          {cancelButtonText && (
```
Comment (shaneeza) [inline packages/form-footer/src/FormFooter.tsx:136]: In props the descriptions says, `Text for the cancel button. A cancel button will appear regardless of whether text is defined.` but in this case it won't show up. We should update that if it's not the case anymore.

File: packages/form-footer/src/FormFooter.tsx [hunk 2]
```diff
@@ -119,60 +114,41 @@ export default function FormFooter({
...
             </Button>
           )}
-          {errorMessage && (
-            <Banner className={bannerStyle} variant="danger">
-              {errorMessage}
-            </Banner>
+          {isComponentType(primaryButton, 'Button') ? (
+            primaryButton
+          ) : (
+            <PrimaryButton {...(primaryButton as PrimaryButtonProps)} />
```
Comment (shaneeza) [inline packages/form-footer/src/FormFooter.tsx:148]: What happens if a random HTML element is passed, say like a `span`?

**Review by shaneeza — CHANGES_REQUESTED** [review]

File: packages/form-footer/src/FormFooter.tsx (hunk 2 above)
Comment (spark3-mdb) [reply packages/form-footer/src/FormFooter.tsx:148]: Ah that's a good catch. I've brought back the code to raise an error once to the `PrimaryButton` component!

File: packages/form-footer/src/FormFooter.tsx (hunk 1 above)
Comment (spark3-mdb) [reply packages/form-footer/src/FormFooter.tsx:136]: README updated!

File: packages/form-footer/README.md [hunk 3]
```diff
@@ -54,7 +54,7 @@ npm install @leafygreen-ui/form-footer
 | Prop             | Type                                | Description                                                                                             | Default  |
 | ---------------- | ----------------------------------- | ------------------------------------------------------------------------------------------------------- | -------- |
 | primaryButton    | `<Button>` or `PrimaryButtonProps`  | The primary (right-most) button. Defined as a `<Button>` element, or as an object with the shape below. |          |
-| cancelButtonText | `string`                            | Text for the cancel button. A cancel button will appear regardless of whether text is defined.          | "Cancel" |
+| cancelButtonText | `string`                            | Text for the cancel button. A cancel button will only appear if this text is defined.                   | "Cancel" |
```
Comment (shaneeza) [inline packages/form-footer/README.md:57]: You'll also have to update it in `FormFooterProps`

File: packages/form-footer/src/PrimaryButton.tsx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import Button from '@leafygreen-ui/button';
+import { once } from 'lodash';
```
Comment (shaneeza) [inline packages/form-footer/src/PrimaryButton.tsx:3]: There's a helper for this inside of `lib`

```
export const consoleOnce = {
  error: once(console.error),
  warn: once(console.warn),
  log: once(console.log),
};
```

Comment (spark3-mdb) [reply packages/form-footer/src/PrimaryButton.tsx:3]: Didn't know this! Thanks! :)

**Review by shaneeza — APPROVED** [review]

**Pull request merged**